src/group.rs

```diff
@@ -157,22 +162,35 @@ impl GroupOps for crate::IronOxide {
         rt.block_on(group_api::list(self.device.auth(), None))
     }
 
-    fn group_create(&self, opts: &GroupCreateOpts) -> Result<GroupMetaResult> {
+    fn group_create(&self, opts: &GroupCreateOpts) -> Result<GroupCreateResult> {
         let mut rt = Runtime::new().unwrap();
         let GroupCreateOpts {
             id: maybe_id,
             name: maybe_name,
             add_as_member,
+            members,
             needs_rotation,
         } = opts.clone();
 
+        let modified_add_as_member;
+        if members.contains(self.device.auth().account_id()) {
+            modified_add_as_member = true;
+        } else {
+            modified_add_as_member = add_as_member;
+        }
+        let modified_members: Vec<UserId> = members
```

Comment:
You already know if the member is in the list (because of the contains check). It seems like we could avoid this filter in that case. I'm happy to help you write that code so we can make things more clear if you want.